docarray/array/storage/redis/backend.py

```diff
@@ -0,0 +1,188 @@
+        super()._init_storage()
+
+        if _docs is None:
+            return
+        elif isinstance(_docs, Iterable):
+            self.extend(_docs)
+        elif isinstance(_docs, Document):
+            self.append(_docs)
+
+    def _build_client(self):
+        client = Redis(
+            host=self._config.host,
+            port=self._config.port,
+            **self._config.redis_config,
+        )
+
+        if self._config.flush:
+            client.flushdb()
+
+        if self._config.update_schema:
+            if self._config.index_name in client.execute_command('FT._LIST'):
+                client.ft(index_name=self._config.index_name).dropindex()
+
+        if self._config.flush or self._config.update_schema:
+            schema = self._build_schema_from_redis_config()
+            idef = IndexDefinition(prefix=[self._doc_prefix])
+            client.ft(index_name=self._config.index_name).create_index(
+                schema, definition=idef
+            )
+
+        return client
+
+    def _ensure_unique_config(
+        self,
+        config_root: dict,
+        config_subindex: dict,
+        config_joined: dict,
+        subindex_name: str,
+    ) -> dict:
+        if 'index_name' not in config_subindex:
+            config_joined['index_name'] = (
+                config_joined['index_name'] + '_subindex_' + subindex_name
+            )
+        config_joined['flush'] = False
+        return config_joined
+
+    def _build_schema_from_redis_config(self):
+        index_param = {
+            'TYPE': 'FLOAT32',
+            'DIM': self.n_dim,
+            'DISTANCE_METRIC': self._config.distance,
+        }
+
+        if self._config.method == 'HNSW':
+            index_options = {
+                'M': self._config.m,
+                'EF_CONSTRUCTION': self._config.ef_construction,
+                'EF_RUNTIME': self._config.ef_runtime,
+            }
+            index_param.update(index_options)
+
+        if self._config.method == 'FLAT' and self._config.block_size:
+            index_options = {'BLOCK_SIZE': self._config.block_size}
+            index_param.update(index_options)
+
+        if self._config.initial_cap:
+            index_param['INITIAL_CAP'] = self._config.initial_cap
+        schema = [VectorField('embedding', self._config.method, index_param)]
+
+        if self._config.tag_indices:
+            for index in self._config.tag_indices:
+                schema.append(TextField(index))
+
+        for col, coltype in self._config.columns:
+            schema.append(self._map_column(col, coltype))
+
+        if self._config.index_text:
+            schema.append(TextField('text'))
+        return schema
+
+    def _doc_id_exists(self, doc_id):
+        return self._client.exists(self._doc_prefix + doc_id)
+
+    def _map_embedding(self, embedding: 'ArrayType') -> bytes:
+        if embedding is not None:
+            from ....math.ndarray import to_numpy_array
```

Comment:
I think we try to use abolsute import instead of `...`, so like `from docarray.whatever.math.ndarray import to_numpy_array`, just for readability